drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/ScenarioWidgetComponentCreator.java

```diff
@@ -74,21 +74,26 @@ protected void reset(final ScenarioParentWidget scenarioWidget,
         this.ruleNamesService.call((Collection<String> ruleNames) -> {
             if (!(ruleNames == null || ruleNames.isEmpty())) {
                 ruleNameSelector.setEnabled(true);
-                ruleNames.forEach((r) -> ruleNameSelector.add(makeRuleNameOption(r)));
+                ruleNames.forEach((r) -> ruleNameSelector.add(makeRuleNameOption(getSimpleRuleName(r))));
             } else {
                 ruleNameSelector.setEnabled(false);
             }
         }).getRuleNames(path,
                         scenario.getPackageName());
     }
 
-    Option makeRuleNameOption(final String text) {
+    Option makeRuleNameOption(final String simpleRuleName) {
         final Option o = GWT.create(Option.class);
-        o.setText(text);
-        o.setValue(text);
+        o.setText(simpleRuleName);
+        o.setValue(simpleRuleName);
         return o;
     }
 
+    String getSimpleRuleName(final String fullyQualifiedRuleName) {
+        final int dotIndex = fullyQualifiedRuleName.lastIndexOf(".");
+        return dotIndex > -1 ? fullyQualifiedRuleName.substring(dotIndex + 1) : fullyQualifiedRuleName;
+    }
```

Comment:
@yurloc What a can of worms this is :-(

"First occurrence" works OK where the package names are completely different (or sufficiently different) but not so for child packages; e.g. ```org.pkg.rule1``` and ```org.pkg.child.rule1``` when we'd end up with a rule called ```child.rule2``` which is clearly wrong. The problem is that the index stores fully qualified rule names (as String) but we don't know what portion is the _package_ vs _rule name_. So a fix (with simple Rule Names) for rules containing "```.```" breaks for child packages and a fix for child packages breaks for rules... Perhaps FQRN and a breaking change for 7.x is the best option after all?